server/src/main/java/org/opensearch/search/internal/ContextIndexSearcher.java

```diff
@@ -204,7 +218,15 @@ public Weight createWeight(Query query, ScoreMode scoreMode, float boost) throws
             // createWeight() is called for each query in the tree, so we tell the queryProfiler
             // each invocation so that it can build an internal representation of the query
             // tree
-            ContextualProfileBreakdown<QueryTimingType> profile = profiler.getQueryBreakdown(query);
+            ContextualProfileBreakdown<QueryTimingType> profile;
+            if (searchContext.shouldUseConcurrentSearch()) {
+                long threadId = Thread.currentThread().getId();
+                ConcurrentQueryProfileTree profileTree = profiler.getThreadToProfileTree()
+                    .computeIfAbsent(threadId, k -> new ConcurrentQueryProfileTree());
```

Comment:
Hey @jed326, it won't be an issue in the case you mentioned because multiple slices are sequential execution in the same thread. The profileTree will be construct just like the non-concurrent search case.